lib/ansible/cli/galaxy.py

```diff
@@ -320,7 +320,10 @@ def server_config_def(section, key, required):
                       ('auth_url', False)]
 
         config_servers = []
-        for server_key in (C.GALAXY_SERVER_LIST or []):
+
+        # Need to filter out empty strings or non truthy values as an empty server list env var is equal to [''].
+        server_list = [s for s in C.GALAXY_SERVER_LIST or [] if s]
```

Comment:
If `server_list` is set to '' in the ansible.cfg that turns into a truthy string (`"''"`), but that's not specific to the galaxy or something that's changed with this PR.